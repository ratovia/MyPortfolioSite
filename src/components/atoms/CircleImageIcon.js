import React from "react"
import styled from "styled-components"

const CircleImage = styled.img`
  border-radius: 50%;
  background-color: white;
`
export const CircleImageIcon = ({ imageUrl, size }) => {
  return (
    <>
      {(() => {
        if (size === "large") {
          return (
            <CircleImage
              src={imageUrl}
              alt="アイコン"
              width="240px"
              height="240px"
            ></CircleImage>
          )
        } else if (size === "small") {
          return (
            <CircleImage
              src={imageUrl}
              alt="アイコン"
              width="48px"
              height="48px"
            ></CircleImage>
          )
        } else {
          return (
            <CircleImage
              src={imageUrl}
              alt="アイコン"
              width="200px"
              height="200px"
            ></CircleImage>
          )
        }
      })()}
    </>
  )
}
