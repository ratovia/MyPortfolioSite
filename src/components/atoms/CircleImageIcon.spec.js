import React from "react"
import renderer from "react-test-renderer"
import javascriptIcon from "../../stories/assets/javascript_icon.png"
import ratoviaIcon from "../../stories/assets/ratovia_icon.jpg"
import reactIcon from "../../stories/assets/react_icon.jpg"

import { CircleImageIcon } from "./CircleImageIcon"
describe("CircleImageIcon", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<CircleImageIcon imageUrl={javascriptIcon} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer
      .create(<CircleImageIcon imageUrl={ratoviaIcon} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer
      .create(<CircleImageIcon imageUrl={reactIcon} size="large" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer
      .create(<CircleImageIcon imageUrl={reactIcon} size="medium" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer
      .create(<CircleImageIcon imageUrl={reactIcon} size="small" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
