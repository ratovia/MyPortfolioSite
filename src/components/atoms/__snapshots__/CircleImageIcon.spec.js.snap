// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleImageIcon renders correctly 1`] = `
<img
  alt="アイコン"
  className="sc-bdvvaa hVPPTo"
  height="200px"
  src="test-file-stub"
  width="200px"
/>
`;

exports[`CircleImageIcon renders correctly 2`] = `
<img
  alt="アイコン"
  className="sc-bdvvaa hVPPTo"
  height="200px"
  src="test-file-stub"
  width="200px"
/>
`;

exports[`CircleImageIcon renders correctly 3`] = `
<img
  alt="アイコン"
  className="sc-bdvvaa hVPPTo"
  height="240px"
  src="test-file-stub"
  width="240px"
/>
`;

exports[`CircleImageIcon renders correctly 4`] = `
<img
  alt="アイコン"
  className="sc-bdvvaa hVPPTo"
  height="200px"
  src="test-file-stub"
  width="200px"
/>
`;

exports[`CircleImageIcon renders correctly 5`] = `
<img
  alt="アイコン"
  className="sc-bdvvaa hVPPTo"
  height="48px"
  src="test-file-stub"
  width="48px"
/>
`;
